SU2_CFD/src/drivers/CDiscAdjSinglezoneDriver.cpp

```diff
@@ -443,56 +443,93 @@ void CDiscAdjSinglezoneDriver::Print_DirectResidual(RECORDING kind_recording){
   /*--- This routine should be moved to the output, once the new structure is in place ---*/
   if ((rank == MASTER_NODE) && (kind_recording == MainVariables)){
 
-    switch (config->GetKind_Solver()) {
-
-    case DISC_ADJ_EULER: case DISC_ADJ_NAVIER_STOKES: case DISC_ADJ_RANS:
-    case DISC_ADJ_INC_EULER: case DISC_ADJ_INC_NAVIER_STOKES: case DISC_ADJ_INC_RANS:
-    case DISC_ADJ_FEM_EULER : case DISC_ADJ_FEM_NS : case DISC_ADJ_FEM_RANS :
-      cout << "log10[U(0)]: "   << log10(solver[FLOW_SOL]->GetRes_RMS(0))
-           << ", log10[U(1)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(1))
-           << ", log10[U(2)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(2)) << "." << endl;
-      cout << "log10[U(3)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(3));
-      if (geometry->GetnDim() == 3) cout << ", log10[U(4)]: " << log10(solver[FLOW_SOL]->GetRes_RMS(4));
-      cout << "." << endl;
-      if ( config->GetKind_Turb_Model() != NONE && !config->GetFrozen_Visc_Disc()) {
-        cout << "log10[Turb(0)]: "   << log10(solver[TURB_SOL]->GetRes_RMS(0));
-        if (solver[TURB_SOL]->GetnVar() > 1) cout << ", log10[Turb(1)]: " << log10(solver[TURB_SOL]->GetRes_RMS(1));
-        cout << "." << endl;
-      }
-      if (config->GetWeakly_Coupled_Heat()){
-        cout << "log10[Heat(0)]: "   << log10(solver[HEAT_SOL]->GetRes_RMS(0)) << "." << endl;
-      }
-      if ( config->AddRadiation()) {
-        cout <<"log10[E(rad)]: " << log10(solver[RAD_SOL]->GetRes_RMS(0)) << endl;
-      }
-      break;
-
-    case DISC_ADJ_FEM:
+    const unsigned short fieldWidth = 12;
+    PrintingToolbox::CTablePrinter RMSTable(&std::cout);
+
+    // 0. Add colum names, addVals=0=false
+    // 1. Add residuals, addVals=1=true
+    for (int addVals = 0; addVals < 2; addVals++) {
+      switch (config->GetKind_Solver()) {
+
+      case DISC_ADJ_EULER: case DISC_ADJ_NAVIER_STOKES: case DISC_ADJ_RANS:
+      case DISC_ADJ_INC_EULER: case DISC_ADJ_INC_NAVIER_STOKES: case DISC_ADJ_INC_RANS:
+      case DISC_ADJ_FEM_EULER : case DISC_ADJ_FEM_NS : case DISC_ADJ_FEM_RANS :
```

Comment:
It would be a good time to replace this for something based on `config->GetFluidProblem()`

Reply:
I now took this very `config->GetFluidProblem()` together with `|| config->GetFEMSolver()`. Those also include the non-DISC_ADJ KindSolvers. I am not sure whether it makes sense there to adapt those functions with an `config->GetFluidProblem(bool isDISC_ADJ = false)` that specifically checks for that.

Reply:
Awesome. It should not be necessary to make that check.